### DISCUSSION TITLE
View all next up

### DISCUSSION THREAD

**Discussion by Alliyance** [jellyfin-archive/jellyfin-roku-legacy#2009] [Q&A] [answered]
Is that normal? First time I see this.

Roku version 2.2.1
![image](https://github.com/user-attachments/assets/00b53a74-f2f7-4465-8f75-cf3bbc36767c)

**Comment by jimdogx** [comment, marked as answer]
It's a new feature that has been available on the Web Client for a long time that is now on the Roku Client.  Admittedly, the UI that shows you all of your Next Up shows needs work, but it's a start.

**Reply by Alliyance** [reply]
Why not leave all next up shows on the same line. I just find the blue folder ugly. I imagine they will change that in the near future.

**Reply by jimdogx** [reply]
A couple of reasons:

1. I believe there is a limit to how many items the Roku can show at one time.  One solution is to load "on the fly" as a user is scrolling, but then you have a slight delay (plus maybe some flickering).

2. The Roku client usually tries to mimic the Web Client as the gold standard.  The Web Client only shows the most recent 12 and then you have to hit "Next Up" to see the rest.

I think a future version of the app could add a new setting that allows user to turn off that feature (so you'd never see any next up items that aren't on the home screen, just like before).  And/Or the icon could be cleaned up to show a collage (just like "Genres" does on the Movie library).

**Reply by Alliyance** [reply]
Well explained, thanks!
